tests/composite-proofs/blind-signature.spec.ts

```diff
@@ -23,63 +23,70 @@ describe('Getting a blind signature, i.e. signature where signer is not aware of
     const messageCount = 5;
 
     const label = stringToBytes('My sig params in g1');
-    const params = SignatureParamsG1.generate(messageCount, label);
+    const params = SignatureParams.generate(messageCount, label);
 
     // Signers keys
-    const keypair = KeypairG2.generate(params);
+    const keypair = KeyPair.generate(params);
     const sk = keypair.secretKey;
     const pk = keypair.publicKey;
+    const h = generateRandomG1Element();
 
     // Prepare messages that will be blinded (hidden) and known to signer
     const blindedMessages = new Map();
     const knownMessages = new Map();
+    const blindings = new Map();
 
     // User wants to hide messages at indices 0 and 2 from signer
     const blindedIndices: number[] = [];
     blindedIndices.push(0);
-    blindedMessages.set(0, stringToBytes('my-secret'));
+    blindedMessages.set(0, encodeMessageForSigning(stringToBytes('my-secret')));
     blindedIndices.push(2);
-    blindedMessages.set(2, stringToBytes('my-another-secret'));
+    blindedMessages.set(2, encodeMessageForSigning(stringToBytes('my-another-secret')));
+
+    // Signer will know these message
+    knownMessages.set(1, encodeMessageForSigning(stringToBytes('John Smith')));
+    knownMessages.set(3, encodeMessageForSigning(stringToBytes('john.smith@emample.com')));
+    knownMessages.set(4, encodeMessageForSigning(stringToBytes('New York')));
 
     // Blind signature request will contain a Pedersen commitment, and it can be given a blinding of choice
     // or it can generate on its own.
-    const [blinding, request] = BlindSignatureG1.generateRequest(blindedMessages, params, true);
-
-    expect(request.blindedIndices).toEqual(blindedIndices);
+    let blinding, request;
+    if (isPS()) {
+      [blinding, request] = BlindSignature.generateRequest(
+        blindedMessages,
+        blindings,
+        params,
+        h,
+        void 0,
+        knownMessages
+      );
+    } else if (isBBSPlus()) {
+      [blinding, request] = BlindSignature.generateRequest(blindedMessages, params, false, void 0, knownMessages);
```

Comment:
On PS doesn't support `encodeMessages` but other 2 do it should be kept true as before.